config/routes.rb

```diff
@@ -394,10 +392,18 @@
       end
 
       scope '/references' do
-        get '/start' => 'decoupled_references#start', as: :decoupled_references_start
+        get '/start' => 'decoupled_references/base#start', as: :decoupled_references_start
+
+        get '/type' => 'decoupled_references/type#new', as: :decoupled_references_new_type
+        post '/update-type' => 'decoupled_references/type#create', as: :decoupled_references_create_type
+
+        get '/name/:id' => 'decoupled_references/name#new', as: :decoupled_references_new_name
+        post '/name/:id' => 'decoupled_references/name#create', as: :decoupled_references_create_name
```

Comment:
Making the `post` a `patch` means we use the correct HTTP verb to get here, and provides another hint that we're updating an existing resource at this endpoint. Also, an `edit/update` action pair is probably more accurate than `new/create`.

Reply:
Okay cool. We need additional actions to go back and edit/update their original choice, then redirect to the unsubmitted review page. I think for now it's probs best to leave it at this until the flow is working. Then I'll come back and make the requested changes. I've added them to the card.